pydantic_ai_slim/pydantic_ai/models/__init__.py

```diff
@@ -710,7 +710,27 @@ def infer_model(model: Model | KnownModelName | str) -> Model:  # noqa: C901
         )
         provider = 'google-vertex'
 
-    if provider == 'cohere':
+    if provider == 'gateway':
+        # TODO(Marcelo): We need to try..except ValueError.
+        forward_provider, model_name = model_name.split('/', 1)
+        if forward_provider == 'openai':
+            from .openai import OpenAIChatModel, OpenAIResponsesModel
+
+            # TODO(Marcelo): We need to try..except ValueError.
+            endpoint, model_name = model_name.split('/', 1)
+            openai_model_cls = OpenAIResponsesModel if endpoint == 'responses' else OpenAIChatModel
+            return openai_model_cls(model_name, provider='gateway')
```

Comment:
This means we'll be sending the regular model name, without any provider-identifying prefix -- how will the gateway know what API to route to?

Reply:
Yes, it was my bad. I've fixed it.